src/shogun2-util/src/main/java/de/terrestris/shogun2/util/http/HttpUtil.java

```diff
@@ -486,6 +488,138 @@ private static Response postParams(HttpPost httpRequest,
+	 *
+	 * @param uriString
+	 * @return
+	 * @throws HttpException
+	 * @throws URISyntaxException
+	 */
+	public static Response put(String uriString, String username, String password) throws URISyntaxException, HttpException {
+		return putBody(new HttpPut(uriString), StringUtils.EMPTY, ContentType.APPLICATION_JSON, username, password);
+	}
+
+	/**
+	 * Performs an HTTP PUT on the given URL.
+	 * (without BasicAuth)
+	 *
+	 * @param uriString
+	 * @param body
+	 * @param contentType
+	 * @return
+	 * @throws URISyntaxException
+	 * @throws HttpException
+	 */
+	public static Response put(String uriString, String body, ContentType contentType) throws URISyntaxException, HttpException{
+		return putBody(new HttpPut(uriString), body, contentType, null, null);
+	}
+
+	/**
+	 * Performs an HTTP PUT on the given URL.
+	 *
+	 * @param uriString
+	 * @param body
+	 * @param contentType
+	 * @param username The Basic authentication username. No basic auth if null.
+	 * @param password The Basic authentication password. No basic auth if null.
+	 *
+	 * @return The HTTP response as Response object.
+	 * @throws URISyntaxException
+	 * @throws HttpException
+	 */
+	public static Response put(String uriString, String body, ContentType contentType, String username, String password) throws URISyntaxException, HttpException{
+		return putBody(new HttpPut(uriString), body, contentType, username, password);
+	}
+
+	/**
+	 * Performs an HTTP PUT on the given URL.
+	 * without BasicAuth
+	 *
+	 * @param uri
+	 * @param body
+	 * @param contentType
+	 * @return
+	 * @throws URISyntaxException
+	 * @throws HttpException
+	 */
+	public static Response put(URI uri, String body, ContentType contentType) throws URISyntaxException, HttpException{
+		return putBody(new HttpPut(uri), body, contentType, null, null);
+	}
+
+	/**
+	 * Performs an HTTP PUT on the given URL.
+	 *
+	 * @param uri
+	 * @param body
+	 * @param contentType
+	 * @param username The Basic authentication username. No basic auth if null.
+	 * @param password The Basic authentication password. No basic auth if null.
+	 *
+	 * @return The HTTP response as Response object.
+	 * @throws URISyntaxException
+	 * @throws HttpException
+	 */
+	public static Response put(URI uri, String body, ContentType contentType, String username, String password) throws URISyntaxException, HttpException{
+		return putBody(new HttpPut(uri), body, contentType, username, password);
+	}
+
+	/**
+	 *
+	 * @param httpRequest
+	 * @param body
+	 * @param contentType
+	 * @param username
+	 * @param password
+	 * @return The HTTP response as Response object.
+	 * @throws URISyntaxException
+	 * @throws HttpException
+	 */
+	private static Response putBody(HttpPut httpRequest, String body, ContentType contentType, String username, String password) throws URISyntaxException, HttpException {
+		StringEntity stringEntity = new StringEntity(body, contentType);
+		stringEntity.setChunked(true);
+		httpRequest.setEntity(stringEntity);
```

Comment:
Maybe you can introduce a condition here that checks for a not-null ContentType.